### PR TITLE
Bug 271376:[dotnet] protected member not being published

### PR DESCRIPTION
**Pull request by v-haiboz**

### CODE REVIEW

File: ECMA2Yaml/ECMAHelper/Models/ECMAStore.cs [hunk 1]
```diff
@@ -550,6 +550,17 @@ private void PopulateMonikers()
                             if (_frameworks.DocIdToFrameworkDict.ContainsKey(m.DocId))
                             {
                                 m.Monikers = new HashSet<string>(_frameworks.DocIdToFrameworkDict[m.DocId]);
+                                if (m.Signatures.IsProtected)
+                                {
+                                    //Filter out moniker of members that are in public sealed class;
+                                    m.Monikers = m.Monikers.Where(p =>
```
Comment (TianqiZhang) [inline ECMA2Yaml/ECMAHelper/Models/ECMAStore.cs]: this piece of logic seems duplicated with the `IsPublishSealedClass` property. Can we make it a shared method?

Comment (v-haiboz) [reply ECMA2Yaml/ECMAHelper/Models/ECMAStore.cs]: it can't be shared, have different logic

Comment (TianqiZhang) [inline ECMA2Yaml/ECMAHelper/Models/ECMAStore.cs]: you can get all the versions that are public sealed, and then intersect m.Moniker with it.

File: ECMA2Yaml/ECMAHelper/Models/ECMAStore.cs [hunk 2]
```diff
@@ -550,6 +550,13 @@ private void PopulateMonikers()
                             if (_frameworks.DocIdToFrameworkDict.ContainsKey(m.DocId))
                             {
                                 m.Monikers = new HashSet<string>(_frameworks.DocIdToFrameworkDict[m.DocId]);
+                                if (m.Signatures.IsProtected)
+                                {
+                                    //Filter out moniker of members that are in public sealed class;
+                                    var publishSealedClasses = m.Parent.Signatures.GetPublishSealedClasses()?.Select(s => s.Monikers)?.ToList();
```
Comment (TianqiZhang) [inline ECMA2Yaml/ECMAHelper/Models/ECMAStore.cs]:
```suggestion
                                    var publishSealedClasses = m.Parent.Signatures.GetPublishSealedClasses()?.SelectMany(s => s.Monikers).ToList();
```

File: ECMA2Yaml/ECMAHelper/Models/ECMAStore.cs [hunk 3]
```diff
@@ -550,6 +550,13 @@ private void PopulateMonikers()
...
+                                    publishSealedClasses?.ForEach(p => m.Monikers = m.Monikers.Except(p).ToHashSet());
```
Comment (TianqiZhang) [inline ECMA2Yaml/ECMAHelper/Models/ECMAStore.cs]:
```suggestion
                                     m.Monikers = m.Monikers.Except(publishSealedClasses);
```

**Pull request closed without merging**